src/lib.rs

```diff
@@ -1178,4 +1259,70 @@ mod tests {
         assert_eq!(hue, hue_after_mixing(Color::graytone(0.7)));
         assert_eq!(hue, hue_after_mixing(Color::white()));
     }
+
+    #[test]
+    fn color_scale_add_preserves_ordering() {
+        let mut color_scale = ColorScale::empty();
+
+        color_scale
+            .add_stop(Color::red(), Fraction::from(0.5))
+            .add_stop(Color::gray(), Fraction::from(0.0))
+            .add_stop(Color::blue(), Fraction::from(1.0));
+
+        assert_eq!(color_scale.color_stops.get(0).unwrap().color, Color::gray());
+        assert_eq!(color_scale.color_stops.get(1).unwrap().color, Color::red());
+        assert_eq!(color_scale.color_stops.get(2).unwrap().color, Color::blue());
```

Comment:
Do we want the `color_stops` field to be public? This way, you can not guarantee the "list of color stops always ordered" invariant to be true.

I think, I'd rather make the field private. This way, you can not keep this test, but it's only testing an implementation detail anyway(?).

Reply:
I also think it's better, so the color stops are always ordered. It could help to combine / merge color scales in the future.

This field is already private, but I can access it because the tests are located at the same module.

Reply:
:+1: for some reason I thought it was public.